kolibri/core/assets/src/core-actions.js

```diff
@@ -83,6 +83,78 @@ function _sessionState(data) {
   return state;
 }
 
+const masteryLoggingMap = {
+  id: 'pk',
+  start_timestamp: 'start_timestamp',
+  completion_timestamp: 'completion_timestamp',
+  end_timestamp: 'end_timestamp',
+  mastery_criterion: 'mastery_criterion',
+  complete: 'complete',
+  responsehistory: 'responsehistory',
+};
+
+function _masteryLoggingState(data) {
+  const state = {};
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    state[key] = data[masteryLoggingMap[key]];
+  });
+  return state;
+}
+
+function _masteryLogModel(store) {
+  const mapping = {};
+  const masteryLog = store.state.core.logging.mastery;
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    mapping[masteryLoggingMap[key]] = masteryLog[key];
+  });
+  mapping.summarylog = store.state.core.logging.summary.id;
```

Comment:
Basically, we set the summarylog id on the masterylog, based on the current summarylog state.

All other fields are set in the `forEach` above directly from the masterylog state.